cosmwasm/contracts/token-bridge/src/contract.rs

```diff
@@ -1378,6 +1428,27 @@ fn handle_initiate_transfer_native_token(
     // we store here just in case the token is transferred out before it's attested
     let asset_address = TokenId::Bank { denom }.store(deps.storage)?;
 
+    let decimals = cfg.native_decimals;
+    let multiplier = 10u128.pow((max(decimals, 8u8) - 8u8) as u32);
+
+    // chop off dust
+    amount = Uint128::new(
+        amount
+            .u128()
+            .checked_sub(amount.u128().checked_rem(multiplier).unwrap())
+            .unwrap(),
+    );
```

Comment:
might be a little cleaner like this:
```suggestion
  amount = amount - amount % Uint128::new(multiplier);
```
`amount` is of type `UInt128` so its operations are all checked (e.g. https://github.com/CosmWasm/cosmwasm/blob/main/packages/std/src/math/uint128.rs#L331-L341) and the error messages are handled better than what `unwrap()` gives. In any case, the rest of the code looks like this too, so for now we can just use this version and make a note of doing a refactoring round later.